packages/shared/src/components/squads/SharePostBar.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { FormEvent, ReactElement, useState } from 'react';
+import classNames from 'classnames';
+import { useMutation } from 'react-query';
+import { ProfilePicture } from '../ProfilePicture';
+import { Button, ButtonSize } from '../buttons/Button';
+import { useAuthContext } from '../../contexts/AuthContext';
+import useSidebarRendered from '../../hooks/useSidebarRendered';
+import { getPostByUrl, Post } from '../../graphql/posts';
+
+export interface NewSquadPostProps {
+  post?: Post;
+  url?: string;
+}
+
+interface SharePostBarProps {
+  className?: string;
+  onNewSquadPost?: (props?: NewSquadPostProps) => void;
+}
+
+function SharePostBar({
+  className,
+  onNewSquadPost,
+}: SharePostBarProps): ReactElement {
+  const [url, setUrl] = useState('');
+  const { mutateAsync: getPost } = useMutation(getPostByUrl, {
+    onSuccess: (post) => {
+      onNewSquadPost({ post });
+    },
+    onError: () => {
+      onNewSquadPost({ url });
+    },
+  });
+  const { user } = useAuthContext();
+  const { sidebarRendered } = useSidebarRendered();
+  const onSubmit = async (e: FormEvent) => {
+    e.preventDefault();
+    await getPost(url);
+  };
+
+  return (
+    <form
+      onSubmit={onSubmit}
+      className={classNames(
+        'flex flex-wrap items-center rounded-16 typo-callout border overflow-hidden',
+        'bg-theme-float focus-within:border-theme-divider-primary border-theme-divider-tertiary hover:border-theme-divider-primary',
+        className,
+      )}
+    >
+      <ProfilePicture
+        className="m-3"
+        user={user}
+        size="large"
+        nativeLazyLoading
+      />
+      <input
+        type="url"
+        name="share-post-bar"
+        className="flex-1 pl-1 outline-none bg-theme-bg-transparent text-theme-label-primary focus:placeholder-theme-label-quaternary hover:placeholder-theme-label-primary typo-callout"
+        placeholder="Enter link to share"
+        onInput={(e) => setUrl(e.currentTarget.value)}
+        required
```

Comment:
Should be non-required as user can simply press "Post" as empty field.

Reply:
Good catch!